### PR TITLE
Add Function to Store ErrorLogs

### PR DESCRIPTION
**Pull request by seemavora**

### CODE REVIEW

File: api/routes/ErrorLog.js [hunk 1]
```diff
@@ -3,21 +3,28 @@ const router = express.Router();
 const ErrorLog = require('../models/ErrorLog');
 const { OK, BAD_REQUEST } = require('../constants').STATUS_CODES;
 
-router.post('/addErrorLog', (req, res) => {
+async function infoErrorLog(req){
+  let errorSaved =true;
```
Comment (evanugarte) [inline api/routes/ErrorLog.js]:
```suggestion
  let errorSaved = true;
```

File: api/routes/ErrorLog.js [hunk 2]
```diff
@@ -3,21 +3,28 @@ const router = express.Router();
...
-    if (error) {
-      res.sendStatus(BAD_REQUEST);
-    } else {
-      res.sendStatus(OK);
-    }
-  });
+router.post('/addErrorLog', async (req, res) => {
+  if (!await infoErrorLog(req)) {
+    return res.sendStatus(BAD_REQUEST);
+  }
```
Comment (evanugarte) [inline api/routes/ErrorLog.js]:
```suggestion
  } else {
```

File: api/util/errorLog-function.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
+const router = express.Router();
+const ErrorLog = require('../models/ErrorLog');
+
+async function infoErrorLog(req){
```
Comment (evanugarte) [inline api/util/errorLog-function.js:5]: can you call this function something descriptive

```suggestion
async function addErrorLog(req){
```

File: api/util/errorLog-function.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const express = require('express');
```
Comment (evanugarte) [inline api/util/errorLog-function.js:1]: why is this here?

File: api/util/errorLog-function.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+const router = express.Router();
```
Comment (evanugarte) [inline api/util/errorLog-function.js:2]: why is this here?

File: api/util/errorLog-function.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    errorTime: req.body.errorTime,
+    apiEndpoint: req.body.apiEndpoint,
+    errorDescription: req.body.errorDescription
+  });
+  await newError.save()
+    .catch(_ => {
+      errorSaved = false;
+    });
+  return errorSaved;
+}
```
Comment (evanugarte) [inline api/util/errorLog-function.js:18]: add a line between end of function and export line
```suggestion
}

```

File: api/util/errorLog-function.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+module.exports = {infoErrorLog, router};
```
Comment (evanugarte) [inline api/util/errorLog-function.js:19]: we dont use the router variable anywhere, remove it

File: api/util/errorLog-function.js (hunk 3 above)
Comment (evanugarte) [inline api/util/errorLog-function.js:5]: it would be helpful to add some documentation for what this function does (what the param is, what it returns)

see the [file you wrote](https://github.com/SCE-Development/Core-v4/blob/dev/src/APIFunctions/ErrorLog.js) for an example.

**Comment by seemavora** [comment]
> We can now use the function to start storing errors to mongodb. The next step would be to integrate the function in the various APIs we have

In order to do this do I call the function wherever there is a console.log statement?

File: api/util/errorLog-function.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    errorDescription: req.body.errorDescription
+  });
+  await newError.save()
+    .catch(_ => {
+      errorSaved = false;
+    });
+  return errorSaved;
+}
+/**
+ * Add a new error.
```
Comment (evanugarte) [inline api/util/errorLog-function.js:18]: youre supposed to paste the documentation **above the function**. Look inside of any file in [`src/APIFunctions`](https://github.com/SCE-Development/Core-v4/tree/dev/src/APIFunctions)

**Comment by hsorathia** [comment]
Traivs CI pr tests passed but branch didn't, so re-run branch test and let's see what happens
